endpoints/v2/manifest.py

```diff
@@ -270,55 +271,57 @@ def delete_manifest_by_digest(namespace_name, repo_name, manifest_ref):
     Note: there is no equivalent method for deleting by tag name because it is
     forbidden by the spec.
     """
-    repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
-    if repository_ref is None:
-        raise NameUnknown()
+    with db_disallow_replica_use():
```

Comment:
Transaction without the DB side; I want to ensure we don't hit the read replica for gets-then-use, but we don't need the full safety (and overhead) of a transaction because if we find an inconsistency, we can just raise a user error. What I don't want to happen is us try to read something we just wrote as part of the push protocol.